src/foam/doc/ModelBrowser.js

```diff
@@ -0,0 +1,238 @@
+foam.CLASS({
+  package: 'foam.doc',
+  name: 'ModelBrowser',
+  extends: 'foam.u2.Element',
+  documentation: 'Show UML & properties for passed in models',
+
+  requires: [
+    'foam.doc.DocBorder',
+    'foam.doc.ClassList',
+    'foam.doc.SimpleClassView',
+    'foam.doc.UMLDiagram',
+    'foam.nanos.boot.NSpec'
+  ],
+
+  imports: [
+    'nSpecDAO'
+  ],
+
+  exports: [
+    'showOnlyProperties',
+    'showInherited'
+  ],
+
+  properties: [
+    {
+      name: 'models',
+      value: []
+    },
+    {
+      name: 'showOnlyProperties',
+      value: true
+    },
+    {
+      name: 'showInherited',
+      value: false
+    }
+  ],
+
+  css: `
+    ^ {
+      display: flow-root;
+      height: auto;
+      width: 700px;
+      margin: 20px;
+    }
+    ^ .foam-doc-UMLDiagram{ 
+      width: 700px;
+      margin: 0;
+      margin-bottom: 20px;
+    }
+    ^ .foam-doc-UMLDiagram canvas{
+      width: 700px;
+    }
+    ^ .foam-u2-view-TableView-foam-doc-PropertyInfo{
+      width: 700px;
+      float: left;
+      margin-top: 20px;
+      margin-bottom: 30px;
+    }
+    ^ .net-nanopay-ui-ActionView-printPage{
+      margin-top: 20px;
+    }
+    @media print{
+      .foam-u2-view-TableView-th-editColumns{
+        display: none;
+      }
+      ^ .net-nanopay-ui-ActionView-printPage{
+        display: none;
+      }
+    }
+  `,
+
+  methods: [
+    function initE(){
+      this.SUPER();
+      var self = this;
+
+      this.nSpecDAO.select().then(function(d){
+        d.array.forEach(function(n){
```

Comment:
You could just pass this function to select() directly and skip the extra step/nesting.